wrappers/core18.go

```diff
@@ -431,7 +431,7 @@ func undoSnapdServicesOnCore(s *snap.Info, sysd systemd.Systemd) error {
 	return nil
 }
 
-func writeSnapdUserServicesOnCore(s *snap.Info, inter Interacter) error {
+func writeSnapdUserServicesOnCore(s *snap.Info, opts *AddSnapdSnapServicesOptions, inter Interacter) error {
```

Comment:
Nitpick: The only consideration I have is whether we want to pass the entire Options structure when we're only interested in whether we preseed or not. However it's not always super nice either to pass a bool. Just a consideration though and not something you need to change.

Reply:
I chose this signature to match the style of the existing `writeSnapdToolingMountUnit` function. As it's an internal function, it doesn't really make a difference from a refactoring point of view.

One point in favour of the struct is that if a second boolean option is added, we don't have to worry that the wrong option is being passed to the function.